lib/components/sprite_stack_component.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:ui';
+
+import '../sprite.dart';
+import '../spritesheet.dart';
+import 'component.dart';
+
+class SpriteStackComponent extends PositionComponent {
+  SpriteSheet spriteSheet;
+
+  static final _sprites = <String, Map<double, Sprite>>{};
+
+  @override
+  set angle(double value) {
+    if (_sprites[spriteSheet.imageName][value] == null) {
+      _generated = false;
+    }
+    super.angle = value;
+  }
+
+  SpriteStackComponent(
+      {String imageName,
+      int textureWidth,
+      int textureHeight,
+      int columns,
+      int rows}) {
+    if (_sprites[imageName] == null) {
+      _sprites[imageName] = new Map();
+    }
+    x = 100;
+    y = 100;
+    width = textureWidth.toDouble();
+    height = textureHeight.toDouble();
+    spriteSheet = SpriteSheet(
+        imageName: imageName,
+        textureWidth: textureWidth,
+        textureHeight: textureHeight,
+        columns: columns,
+        rows: rows);
+  }
+
+  @override
+  void render(Canvas canvas) {
+    if (!loaded()) {
+      return;
+    }
+    prepareCanvas(canvas);
+    _sprites[spriteSheet.imageName][angle]
+        .render(canvas, width: width, height: height);
+  }
+
+  bool _generated = false;
+  bool _generating = false;
+
+  @override
+  bool loaded() {
+    for (var row = 0; row < spriteSheet.rows; row++) {
```

Comment:
You should cache this, so when this is already done loading, the code will node run those fors anymore. I know these fors should be small and will not hurt the CPU very much, but if have a lot of SpriteStackComponent instances on the game, this can make a difference.

Reply:
Yea you are right, if the `_generated` is already true there is no need to go through the images. Will fix it in the next commit